Create terraform.yml

petrolifero opened the pull request:
#4

petrolifero committed:
Create terraform.yml

petrolifero committed:
install terraform

petrolifero committed:
make terraform profile-independent

petrolifero committed:
change curl to wget

petrolifero committed:
format code

dbofmmbt commented on .github/workflows/terraform.yml:
```diff
@@ -0,0 +1,40 @@
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: install Terraform
+      run: wget https://releases.hashicorp.com/terraform/0.15.5/terraform_0.15.5_linux_amd64.zip && unzip terraform_0.15.5_linux_amd64.zip && export PATH=$PATH:$(pwd)
+
```
Wouldn't it be better to use the Hashicorp's github action to install it?

https://github.com/marketplace/actions/hashicorp-setup-terraform

petrolifero replied:
I think that was only to terraform cloud, but is not. so is implemented

dbofmmbt commented on .github/workflows/terraform.yml:
```diff
@@ -0,0 +1,40 @@
+    - name: Terraform Init
+      run: terraform init
+
+    - name: Terraform Format
+      run: terraform fmt -check
+
+    - name: Terraform Plan
+      run: terraform plan
+
+    - name: Terraform Apply
+      if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+      run: terraform apply -auto-approve
```
I see that there's no human revision of the plan right on the PR in the worflow as it is. You can find this step in these references:

- https://learn.hashicorp.com/tutorials/terraform/github-actions?in=terraform/automation#review-actions-workflow
- https://github.com/marketplace/actions/hashicorp-setup-terraform

Basically, they take the output of the `terraform plan` command and show it directly on the PR to be easier to review. Do you think it is worth adding it?

petrolifero replied:
This is great, I will add

petrolifero replied:
the github actions is done, i need to create a token on github. I will do this on monday.

petrolifero committed:
don't install manually terraform

petrolifero committed:
enable post on pull requests

dbofmmbt commented:
> #### Terraform Format and Style ``
> #### Terraform Initialization ``
> #### Terraform Plan ``
> Show Plan
> *Pusher: @petrolifero, Action: `pull_request`*

@petrolifero this comment looks a bit broken. Is it already fixed?

petrolifero commented:
> 
> 
> > #### Terraform Format and Style ``
> > #### Terraform Initialization ``
> > #### Terraform Plan ``
> > Show Plan
> > _Pusher: @petrolifero, Action: `pull_request`_
> 
> @petrolifero this comment looks a bit broken. Is it already fixed?

Is broken. I will see this today (now, in fact)

petrolifero committed:
set id's on steps

the broken comment of github-bot was happen because the template
make reference to each step by id's

petrolifero committed:
change string's quotes

petrolifero committed:
Revert "change string's quotes"

This reverts commit df64c28414978019b7823294eece94f764b8e4ad.

petrolifero committed:
remove color code from terraform plan

The original output of terraform plan is printed using ansi escape codes to colorize
the plan. Github bot do not understand this codes, so they must not happen

petrolifero committed:
duplicate quotes on process.env.PLAN

petrolifero committed:
Revert "duplicate quotes on process.env.PLAN"

This reverts commit 2318fc3d914bb765f715090445ad47595ef6a6ec.

petrolifero committed:
change quotes on env

petrolifero committed:
Revert "change quotes on env"

This reverts commit 8897a252a71c76e81b7b5a8a5f3a1f32216a0886.

petrolifero committed:
delete ```

dbofmmbt committed:
Update terraform.yml

dbofmmbt committed:
Putting more whitespace around lines to see if GH accepts it

dbofmmbt commented:
Whitespace is important to markdown, indeed :laughing:
